app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -848,9 +826,12 @@ class ComposeActivity : BaseActivity(),
         }
     }
 
-    private fun pickMedia(uri: Uri) {
+    private fun pickMedia(uri: Uri, contentInfoCompat: InputContentInfoCompat? = null) {
         withLifecycleContext {
             viewModel.pickMedia(uri).observe { exceptionOrItem ->
+
+                contentInfoCompat?.releasePermission()
```

Comment:
I'm not good at this stuff so I'm not sure why this is moved here.

Reply:
Because the permission should be released after we no longer need it, so I moved it after the upload is complete (or failed).